lib/posix/utsname.c

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2023 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/posix/sys/utsname.h>
+#include <stdio.h>
+#include <string.h>
+
+int uname(struct utsname *name)
```

Comment:
I'm not usually one for adding to the ROM footprint, but it might be significantly simpler to have 1 `static const struct utsname` in this file and then `memcpy()` it to the destination.

It makes sense why we have the array sizes now, so please disregard my earlier Q.

Reply:
do we need a mechanism to detect if hostname is changed if `net_hostname_set_postfix` is enabled, and update that static copy accordingly?

Reply:
Ok then, `memcpy()` + `snprintf()` for the hostname.